### PR TITLE
Use back-end upload if NML is dragged into a read-only view

### LINKED ISSUE #4417: Support NML import in view mode
Simplest way would probably be to use the back-end upload and redirect to the new annotation.

### PR DESCRIPTION
**Pull request by philippotto**
### URL of deployed dev instance (used for testing):
- https://importinview.webknossos.xyz

### Steps to test:
- Open a dataset in view mode (or a tracing of another use)
- drag an NML file into the tracing view
- click "create new tracing" in the imprt modal
- webKnossos should reload and show a new tracing with the imported NML

### Issues:
- fixes #4417 

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- [ ] ~Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~
- [ ] ~Updated [documentation](../blob/master/docs) if applicable~
- [ ] ~Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes~
- [ ] ~Needs datastore update after deployment~
- [X] Ready for review

### CODE REVIEW

File: frontend/javascripts/oxalis/view/layouting/tracing_layout_view.js [hunk 1]
```diff
@@ -167,8 +170,21 @@ class TracingLayoutView extends React.PureComponent<Props, State> {
...
+        data: { nmlFile: files, createGroupForEachFile },
+      });
+      this.props.history.push(`/annotations/${response.annotation.typ}/${response.annotation.id}`);
+    };
+
     return (
-      <NmlUploadZoneContainer onImport={importTracingFiles} isAllowed={isUpdateTracingAllowed}>
+      <NmlUploadZoneContainer
+        onImport={isUpdateTracingAllowed ? importTracingFiles : createNewTracing}
+        isAllowed={isUpdateTracingAllowed}
```
Comment (daniel-wer) [inline frontend/javascripts/oxalis/view/layouting/tracing_layout_view.js]: Maybe rename the flag to `isUpdateAllowed` or so, to better reflect its usage.

**Comment by philippotto** [comment]
> For 2) we could either substitute the dataset name in the nml with the current dataset name or add an additional dataset overwrite parameter for the backend upload.

Sending a parameter to the backend sounds good to me. I prepared a commit for this, but the back-end part doesn't compile, since I don't know enough about scala :man_shrugging: @youri-k Could you have a look at this if you have a moment? The dataset name might also need to be incorporated into the tracing objects, but I'm not sure about that. Feel free to ask me if anything is unclear.

**Comment by fm3** [comment]
I pushed a fix for the backend, and tested that this does indeed work nicely!
Only thing: if you download the tracing later as NML again, the previous dataset name is written to the NML again. I suppose we don’t want that, right? In this case, we should overwrite the dataset name in the parsed tracing also.

**Comment by fm3** [comment]
new version fixes that also :)

**Comment by philippotto** [comment]
@fm3 Thanks for the fix! Works very well! I only noticed one thing: If try to upload multiple NMLs which were downloaded from a hybrid tracing, the back-end throws `Cannot upload multiple volume tracings` (even though, I only dropped the NMLs into the view). Is this an easy fix?

>  1) is probably not that important as we're the only ones uploading proto files.

Yes, since we don't explicitly advertise this, I think the current way is fine.

**Comment by fm3** [comment]
Hmm, not super easy I’m afraid. I’d create a separate issue for that. For now, to upload those nmls the user will have to remove the `<volume>` tag in them

**Comment by philippotto** [comment]
@daniel-wer I think this PR would be ready to be merged. Would you agree/accept?

**Comment by philippotto** [comment]
@fm3 If you accept this, I think we can merge this :)

**Pull request merged**